core-backup/src/main/scala/io/aiven/guardian/kafka/backup/BackupClientInterface.scala

```diff
@@ -55,36 +56,61 @@ trait BackupClientInterface[T <: KafkaClientInterface] {
                                   override val context: kafkaClientInterface.CursorContext
   ) extends ByteStringElement
 
+  case class PreviousState(state: State, previousKey: String)
```

Comment:
These data structures have to be placed within `BackupClientInterface` because they reference `State` which is an abstract type member of the class itself